src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -525,7 +525,10 @@ static int watch_passwords(void) {
                 return log_error_errno(errno, "Failed to allocate directory watch: %m");
 
         if (inotify_add_watch(notify, "/run/systemd/ask-password", IN_CLOSE_WRITE|IN_MOVED_TO) < 0)
-                return log_error_errno(errno, "Failed to add /run/systemd/ask-password to directory watch: %m");
+                if (errno == ENOSPC)
+                        return log_error("Failed to add /run/systemd/ask-password to directory watch: inotify watch limit reached");
```

Comment:
you are eating up the error here entirely. Please use log_error_errno() instead and pass `errno` as before. Note that passing errno has three effects: 

1. %m resolves to the error message you
2. the negative error number is returned, so that it is easy propagate with `return`.
3. if structured logging to the journal is used the error number is included in an ERRNO= field.

In this case we want 2 and 3, only 1 is not needed. Hence, please use log_error_errno() here, even if you don't make use of `%m`.